junos_config documentation missing "update" choice

crab86 opened the pull request:
##### SUMMARY
The current junos_config module documentation only lists "merge","override" and "replace" as update parameter choices. Looking into junos_python.py "update" is another option which is really helpful and in my opinion the most needed option. 

```
update=dict(default='merge', choices=['merge', 'override', 'replace', 'update'])
```

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
junos_config

crab86 committed:
Update junos_config.py

The current junos_config module documentation only lists "merge","override" and "replace" as update parameter choices. Looking into junos_python.py "update" is another option which is really helpful and in my opinion the most needed option.

'''
update=dict(default='merge', choices=['merge', 'override', 'replace', 'update'])
'''

+label: docsite_pr

acozine commented:
Thanks @crab86 for this - to make the PR pass our CI suite, you need to remove line 618 of [this ignore.txt file](https://github.com/ansible/ansible/blob/devel/test/sanity/validate-modules/ignore.txt#L618). Removing that line will get our CI system to run the `validate-modules` test on the `junos_config.py` file, protecting us against regressions.

samccann commented on lib/ansible/modules/network/junos/junos_config.py:
```diff
@@ -113,8 +113,14 @@
+        replaces the existing configuration. The difference comes when the configuration
+        is later committed. This option performs a 'diff' between the new candidate
+        configuration and the existing committed configuration. It then only notifies
+        system processes repsonsible for the changed portions of the configuration, and
```
```suggestion
        system processes responsible for the changed portions of the configuration, and
```

samccann committed:
Fixup Update junos_config.py

The current junos_config module documentation only lists "merge","override" and "replace" as update parameter choices. Looking into junos_python.py "update" is another option which is really helpful and in my opinion the most needed option.

'''
update=dict(default='merge', choices=['merge', 'override', 'replace', 'update'])
'''

Co-Authored-By: crab86 <sgesenhoff@gmail.com>
